File: backend/app/schemas/analytics.py
"""Analytics shapes."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ProductScanStat(BaseModel):
    product_id: UUID
    name: str
    sku: str
    scan_count: int


class LatestScan(BaseModel):
    scanned_at: datetime
    product_id: UUID
    product_name: str
    sku: str
    country: str
    browser: str
    operating_system: str


class AnalyticsResponse(BaseModel):
    scans_today: int
    scans_this_week: int
    most_viewed_products: list[ProductScanStat]
    latest_scans: list[LatestScan]

File: frontend/types/analytics.ts
export interface ProductScanStat {
  product_id: string;
  name: string;
  sku: string;
  scan_count: number;
}

export interface LatestScan {
  scanned_at: string;
  product_id: string;
  product_name: string;
  sku: string;
  country: string;
  browser: string;
  operating_system: string;
}

export interface AnalyticsSummary {
  scans_today: number;
  scans_this_week: number;
  most_viewed_products: ProductScanStat[];
  latest_scans: LatestScan[];
}
